### PR TITLE
[SimpleGoalChecker] Make parameters changeable at runtime

### PR DESCRIPTION
**Pull request by doisyg**
I need to change the **SimpleGoalChecker** params at runtime, so here is a PR activating this feature in a similar fashion as https://github.com/ros-planning/navigation2/pull/2181
Working and tested on my simulation.
If there is an interest I can add to this PR changeable parameters for the 2 others `nav2_controller` plugins: **SimpleProgressChecker** and **StoppedGoalChecker**

### CODE REVIEW

File: nav2_controller/plugins/simple_goal_checker.cpp [hunk 1]
```diff
@@ -105,6 +119,30 @@ bool SimpleGoalChecker::isGoalReached(
...
+SimpleGoalChecker::on_parameter_event_callback(
+  const rcl_interfaces::msg::ParameterEvent::SharedPtr event)
+{
+  for (auto & changed_parameter : event->changed_parameters) {
+    const auto & type = changed_parameter.value.type;
+    const auto & name = changed_parameter.name;
+    const auto & value = changed_parameter.value;
+
+    if (type == ParameterType::PARAMETER_DOUBLE) {
+      if (name == plugin_name_ + ".xy_goal_tolerance") {
```
Comment (SteveMacenski) [inline nav2_controller/plugins/simple_goal_checker.cpp:132]: Wouldn't this be more than just name + parameter? Wouldn't it also be in the controller_server namespace at minimum? This doesn't have its own node. Did you print out the `name` to make sure this full path is valid?

**Comment by SteveMacenski** [comment]
@doisyg pull in master to fix CI

File: nav2_controller/plugins/simple_goal_checker.cpp (hunk 1 above)
Comment (doisyg) [reply nav2_controller/plugins/simple_goal_checker.cpp:132]: Yes, same logic as for NavFn I believe.
If I do a print in the code:
```
[controller_server-7] [WARN] [SimpleGoalChecker]: name: goal_checker.xy_goal_tolerance
[controller_server-7] [WARN] [SimpleGoalChecker]: plugin_name_: goal_checker
```

**Comment by doisyg** [comment]
Should we investigate if it would be useful to use the newly merged ParameterEventsSubscriber class (https://github.com/ros2/rclcpp/pull/829) for these parameters change use cases ?

File: nav2_controller/plugins/simple_goal_checker.cpp (hunk 1 above)
Comment (SteveMacenski) [reply nav2_controller/plugins/simple_goal_checker.cpp:132]: Ok, works for me then, just verifying

**Comment by SteveMacenski** [comment]
Sure! That would make things much more readable than these single massive callback functions of every parameter

**Comment by doisyg** [comment]
@anaelle-sw, once this is merged, can you try to modify it to use ParameterEventsSubscriber and do a new PR on top of it ?

**Comment by SteveMacenski** [comment]
@doisyg I cancelled the debug workflow because I needed the CI time elsewhere since you already had the release build / test pass.

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**